sdk/eventhub/Azure.Messaging.EventHubs/src/Producer/EventDataBatch.cs

```diff
@@ -16,6 +16,9 @@ namespace Azure.Messaging.EventHubs.Producer
     ///
     public sealed class EventDataBatch : IDisposable
     {
+        /// <summary>A flag indicating that the batch is locked, such as when in use during a publish operation.</summary>
+        private bool _locked = false;
```

Comment:
Based on my limited understanding of threading and the .NET memory model, I think that this needs to be marked as `volatile`?

Reply:
Now there is an important catch.   I never remember the correct memory barrier semantics and keyed on the atomic access only.  Reading through the docs, you're absolutely correct.